api/commands/validate.ts

```diff
@@ -1,16 +1,51 @@
 import * as cronParser from "cron-parser";
 import { dataform } from "@dataform/protos";
+import * as fs from "fs";
+import * as path from "path";
 
+const SCHEDULES_JSON_PATH = "schedules.json";
 const EMAIL_REGEX = /^(([^<>()\[\]\.,;:\s@\"]+(\.[^<>()\[\]\.,;:\s@\"]+)*)|(\".+\"))@(([^<>()[\]\.,;:\s@\"]+\.)+[^<>()[\]\.,;:\s@\"]{2,})$/i;
 
+interface IOptions {
+  projectDir?: string;
+}
+
 function validateEmail(email: string) {
   return EMAIL_REGEX.test(email);
 }
 
-export function validateSchedules(schedules: dataform.schedules.SchedulesJSON, compiledGraph: dataform.ICompiledGraph) {
+export function validateSchedulesFile(
+  compiledGraph: dataform.ICompiledGraph,
+  options: IOptions
+): string[] {
+  if (!options || !options.projectDir) {
```

Comment:
this seems odd to me. the user shouldn't ever be calling `validateSchedulesFile` without a`projectDir`. it seems like invalid usage of the API to me, so let's not support it?

Reply:
I got the influence from here: https://github.com/dataform-co/dataform/blob/master/api/commands/query.ts#L41